src/core_functions/aggregate/distributive/functions.json

```diff
@@ -9,19 +9,35 @@
     {
         "name": "arg_min",
         "parameters": "arg,val",
-        "description": "Finds the row with the minimum val. Calculates the arg expression at that row.",
+        "description": "Finds the row with the minimum val. Calculates the non-NULL arg expression at that row.",
         "example": "arg_min(A,B)",
         "type": "aggregate_function_set",
         "aliases": ["argmin", "min_by"]
     },
+    {
+        "name": "arg_min_null",
+        "parameters": "arg,val",
+        "description": "Finds the row with the minimum val. Calculates the arg expression at that row.",
+        "example": "arg_min_null(A,B)",
+        "type": "aggregate_function_set",
+        "aliases": ["argminnull", "min_null_by"]
```

Comment:
I wonder if we actually want to expose this to users and document it or if we want to keep it hidden and used only as optimization. The advantage to that is that we don't need to test corner cases that can be introduced by users but can't be introduced by us, e.g. we would only need to implement `BLOB/BIGINT` sorting keys and wouldn't need to support `NULL` there if we were the ones to generate it exclusively.

I would say at least we shouldn't add these aliases. They exist for `arg_min/arg_max` for compatibility with other systems.

Reply:
I'll remove the aliases, but I can see users wanting them (plus it would make binding them more annoying).